### ISSUE TITLE
New Greasyfork page management discussion

### ISSUE THREAD

**Issue by BigRedBot** [Phr33d0m/NW-Profession-Bot#82]
I just noticed that new options are being added to the 1.10 version that are not being added to this 3.0 version. One of those options that I would like to see implemented is the daily SCA roll option. Is there any way someone could add that option to this version?

**Comment by WloBeb** [comment]
Pull request from you welcome :D.

**Comment by Phr33d0m** [comment]
@BigRedBrent we have "Cycle SCA" which does that (based on your explanation). In any case I have no idea (nor care) what ends up in the greasemonkey releases.

Could you explain a bit better what that opion you're mentioning does and avoid me having to test it out?

**Comment by BigRedBot** [comment]
@WloBeb
Well, if I can get around to familiarize myself with the code necessary to implement it then I may give it a try. Otherwise, I already know someone has already implemented such a feature, and it would be great if that individual who is already familiar with the code could maybe add a pull request.

I mean, nobody hold their breath waiting on me to figure out how to add this option if they could maybe add it themselves.

@Phr33d0m
Well, I have not tested that feature out myself, I only noticed that it was implemented. I imagine that if checked, it would pick a time of day to automatically run the SCA each day, so that you do not have to manually run it yourself.

As far as I know, this option is enabled by default in the greasyfork https://greasyfork.org/en/scripts/7061-neverwinter-gateway-professions-robot version, but I think it would be best to have it disabled by default.

**Comment by Phr33d0m** [comment]
@BigRedBrent yeah the functionality is done and IIRC it works pretty well. Doing it once a day automatically sure seems nice.

**Comment by RottenMind** [comment]
@Phr33d0m, you allways forgot source, but it matters not. This script not work on Greasymonkey, but it work on Tampermonkey.

...and this ,

https://greasyfork.org/en/forum/discussion/2544/x

As you know source is from userscript.org and current script is now reside in "Greasyfork", ;).

Daily SCA function is not completed, but it is safe to use, also there is script what do Full SCA and it works. https://github.com/LaskonSoftware/NWGatewayAutomation 

For SCA I use this simple function. You may check SCA completions from datamodel, turn tutorials off after checked them via datamodel, I did those for full SCA, but as I allready mention function is not ready for use.

I still enjoy rants like ..._we have "Cycle SCA" which does that (based on your explanation). In any case I have no idea (nor care) what ends up in the greasemonkey releases._

/**
- Created by RM on 29.4.2015.
- Runs daily SCA -rolls in GAteway Bot
  */
  function dailySCA() {
  var char, today, thisday, thishour, dailyroll, dateforlastroll;
  char = settings["charcount"];
  today = new Date();
  thisday = today.getDay();
  thishour = today.getHours();
  if (settings["dailySCA"] && chardelay > 10000 \* char && (thishour >= 14 || thishour >= 23)) {
      dailyroll = GM_getValue("dailyswordcoast", 0);
      dateforlastroll = GM_getValue("dateforlastrolls", 0);
      //console.log(thisday, dateforlastroll, dailyroll, chardelay, thishour);
      if (thisday !== dateforlastroll) {
          GM_setValue("dateforlastrolls", thisday);
          GM_setValue("dailyswordcoast", 0);
          dailyroll = 0;
      }
      if (dailyroll < (4 || undefined)) {
          unsafeWindow.location.hash = unsafeWindow.location.hash.replace(/)\/.+/, ')' + "/adventures");
          processSwordCoastDailies();
          dailyroll++;
          GM_setValue("dailyswordcoast", dailyroll);
          GM_getValue("dailyswordcoast", 0);
      }
  }
  }

ps. oops, format is gone... XD

**Comment by Phr33d0m** [comment]
@RottenMind why would I care what ends up in the greasyfork release? Lets talk about it.

First, you decided to completely split up from the ongoing development of the script and post some edited version of it to greasyfork with the pretext of "it's frendlier to users" and whatnot, making it impossible to track and help support users (thus I don't even look at the issues people are reporting there).

Second, why would I (or anyone here that is actually contributing very valuable and useful changes) care about what ends in the greasyfork release where in the end you'll end up editing/removing most of it?

Third, I'm not ranting. What I'm saying (as a developer) is that I'm not going to support/maintain 2 different versions of the same script.

Forth, I don't understand what you mean by "you allways forgot source, but it matters not", to be honest, I care 0% about where this came from. Of course I'm grateful to all the people that started the thing and have improved it over the time (and I believe they've been credited properly), but it has been abandoned and we (most of the contributors/developers section of the script) are parenting it now. We're parenting it to the point that the script no longer resembles even 20% of what we took from the previous authors.

Fifth, from what I can see in the github project graphs - even more people are coming here (or getting the script from here) which means... something ;-).

Sixth, you ended up removing the link to this repo from your greasyfork description page (and actually linking it to @noonereally's repo, which I don't really know why - as he has direct write access here as well, but meh).

Seventh, it seems you've removed all of the current developers and contributors from your script. Not giving credit is against the license, but well... would you care at all?

If you think something's been done wrong then feel free to spit it out, otherwise I doubt there's any sense in following up this discussion.

**Comment by BigRedBot** [comment]
@RottenMind
I must agree with the sentiments expressed by @Phr33d0m, even if I would not express them as bluntly myself. I don't exactly understand why an almost completely different version is needed on the greasyfork.org site. I believe it would be best if that version was replaced by the stable versions from this development. That is what I originally thought was going to be the case.

If that is not going to be the case, then, perhaps a new page should be created on greasyfork.org for the release versions of this project.

**Comment by RottenMind** [comment]
@BigRedBrent, +1.

 @Phr33d0m, it is good that script lives and that was my main point to when I started new fork on Greasyfork and Github -development, xD. It was good that you took over this Github and it was good that we brought people here. @Phr33d0m, we did this, not you.

...and I look forward day when when current Greasyfork -site is no needed, but it seems that it takes time before it happens.

I may remove link to here when ever I want but its not serve my purpose, so link stays and hopefully you get more users here and make script better.

Script still has same licence and same core code, @Phr33d0m try change them and you get rid all old trash you dont like, xD.

About SCA, Multiaccount -support, AH -sales and several other enchantments, I look forward to see your development for them.

Well, I spitted out that I still enjoy look this stuff and it is my purpose make it bigger and better.

**Comment by Phr33d0m** [comment]
> we did this, not you

@RottenMind and? I believe you didn't understand everything I said.

> Script still has same licence

Yes, and you're not respecting it in your copy of the code in greasyfork.

> and same core code

If by "same core code" you mean we've kept the same function naming (and not much more) then yes, otherwise the "core code" you're talking about has been **almost** completely been rewritten.

> I may remove link to here when ever I want

That is perfectly fine. @BigRedBrent would you be able to open a new greasyfork page (as per what you suggested) so we can offer our current releases there? Because it seems we're going to split from @RottenMind's "management".

**Comment by RottenMind** [comment]
@Phr33d0m , your copy code... hahahahaha, bring the code I disrecpect, ;) nd call take down and I will do.

Page is here, https://greasyfork.org/en/scripts/7808-neverwinter-gateway-professions-robot

@BigRedBrent, I wish you give @Phr33d0m to do new Fork page. It is obivius that he like give responsibility to others. 

ps. I will be happy to link on current Fork to yours as I do to Github, ;).

**Comment by Phr33d0m** [comment]
I'll link everyone that I can think off at the moment so everyone can share an opinion on this.

@noonereally @dlebedynskyi @BigRedBrent @WloBeb @mac-nw - guys could you read our conversation with @RottenMind and please tell me what's the best route we can take?

**Comment by RottenMind** [comment]
@Phr33d0m you took over Github devlopment you may keep it, I gave support for it as long it was benefical for main script, where my support stays until that specific fork userbase drains...

@Phr33d0m , @WloBeb , @dlebedynskyi , @BigRedBrent , @mac-nw , @RottenMind, there is storm in waterglass so you must now drink the water and go toilet, it makes you feel better.

Also read this,

https://greasyfork.org/en/forum/discussion/2544/x

@Phr33d0m  wanted take over Github ... but was it mistake support this? It was not mistake, better have multiple developments, I think Phr33d0m just wake up and you all can run with him different route It will not hurt main script... 

So POV is clear.

**Comment by dlebedynskyi** [comment]
Glared throw. Don't really want to dig into all this mess. 

Create a new version on grease monkey. publish it. Another version is called "Mustex_Bunta blah blah blah" - ok fine. they have thier named version already.
this GitHub version is now in good fairly stable dev circle by @Phr33d0m  and @noonereally. 
And actually I was forking @noonereally version first.
To be honest I don't care who is main supporter. But I know one thing at the point when I was looking there were 10 broken grease-monkey releases with bad testing etc. This one was most stable and relieble. 
Other folks want to move away -ok good luck. competition is a good thing. 
Im gonna contribute to this version as long as releases are good. 

so my thing is - update the freaking gm version finally to new release or create a new one. All who want other -> go to other fork or get older version of this script by github history and releases. All other are welcome to contribute by writing code/ fixing bugs/ giving idea.

We are not 5 year olds here that get offended if someone did better job on some part of code.  Keep calm and bee respectful. we are all here for one reason really.

**Comment by BigRedBot** [comment]
Alright. I have never put a release on greasyfork, but it is a convenient platform to release the stable release on, so, I will try to set something up.

I have a lot of friends who would benefit from having a simple platform to update to the latest release version of this script with. As it stands now, I usually have them manually add the script from this site. It will be easier for me to just point them to a greasyfork site for the release versions of this script.

**Comment by BigRedBot** [comment]
I created this link:
https://greasyfork.org/scripts/9812-neverwinter-gateway-professions-robot/

It is pretty simple. I have it set up so that I can just click a button when a new release version is available and it is imported. We could also just make a release version file somewhere on github and I could have the greasyfork site automatically update to that file, and that would be even simpler. :p

**Comment by RottenMind** [comment]
Nice, you may also change its @namespace that it point to you current new home, ;).

https://greasyfork.org/scripts/9812-neverwinter-gateway-professions-robot/

...and  then I can drive main script to "maintenance mode" and drive most new users from it to your page. OFC, I keep look over old script because it still have users, but by time and changes they move to new.

About SCA, it works perfectly, now five days without stucks, ;).

**Comment by Phr33d0m** [comment]
> We are not 5 year olds

@dlebedynskyi I thought that and I thought I could respectfully and calmly talk with @RottenMind but that wasn't possible thus I linked you all about your opinion on the matter. Thanks.

> I created this link:

Thanks @BigRedBrent, I'll update the links accordingly. On a side note I'll rename and close this issue as this got off hand and quite off the initial topic. I'll create a new issue with your initial message.
